### PR TITLE
[RyuJIT] Propagate gtFlags in Vector.Create

### LINKED ISSUE #43569: System.Runtime.Intrinsics.Vector128.Create() bug inside of for loop
### Description
I believe that I have found a regression with the way Vector128<float> structs are created using Vector128.Create(float, float, float, float) when converting from doubles. For example, the following code console writes <1, 2, 3, 4> as expected:

```csharp
double[] array = new[] { 1.0, 2, 3, 4 };

Vector128<float> x = Vector128.Create(Convert.ToSingle(array[0]), Convert.ToSingle(array[1]), Convert.ToSingle(array[2]), Convert.ToSingle(array[3]));

Console.WriteLine(x);
```

However, simply putting the code inside a for loop (or while for that matter) causes a console write of <1, 2, 3, ___2___>:

```csharp
double[] array = new[] { 1.0, 2, 3, 4 };

Vector128<float> x = Vector128<float>.Zero;

for (int i = 0; i < 1; i++)
    x = Vector128.Create(Convert.ToSingle(array[0]), Convert.ToSingle(array[1]), Convert.ToSingle(array[2]), Convert.ToSingle(array[3]));

Console.WriteLine(x);
```

### Configuration
dotnet 5.0.100-rc.2.20479.15
Windows 10 Enterprise 2004 (19041.572)
x64 in ___Release___ mode ... in Debug mode, things work as expected.

### Regression
Yes, this is a regression. I tried this using .NET Core 3.1 (3.1.402) and got the expected output either way.

### Other information
One way to work around this issue is to set the 3rd element again manually using WithElement() like so:

```csharp
x = x.WithElement(3, Convert.ToSingle(array[3]));
```

Hope this is helpful, please let me know if you have any questions.

Thanks!

### PR DESCRIPTION
**Pull request by EgorBo**
Fixes: https://github.com/dotnet/runtime/issues/43569 bug

Current code used to ignore flags of arguments so we could get a tree like this:
```
[000010] ------------                    \--*  HWINTRINSIC simd16 float Create
[000009] ------------                       \--*  LIST      void
[000000] ------------                          +--*  CNS_DBL   float  1.0000000000000000
[000008] ------------                          \--*  LIST      void
[000001] ------------                             +--*  CNS_DBL   float  2.0000000000000000
[000007] ------------                             \--*  LIST      void
[000002] ------------                                +--*  CNS_DBL   float  3.0000000000000000
[000006] --C---------                                \--*  LIST      void
[000005] --C---------                                   \--*  RET_EXPR  float (inl return from call [000015])
```
As you can see `C` (stands for `GTF_CALL`) is not propagated to the top. That's why `fgUpdateInlineReturnExpressionPlaceHolder ` just gave up on the tree and didn't bother scanning it for `RET_EXPR` to replace with actual value.

/cc @dotnet/jit-contrib 

The bug doesn't reproduce on `netcoreapp3.1`. Worth back-porting to 5.0 ?

### CODE REVIEW

File: src/coreclr/src/jit/hwintrinsicarm64.cpp [hunk 1]
```diff
@@ -396,9 +396,8 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
 
                 for (unsigned i = 0; i < sig->numArgs; i++)
                 {
-                    tmp        = gtNewArgList(impPopStack().val);
-                    tmp->gtOp2 = op1;
-                    op1        = tmp;
+                    tmp = new (this, GT_LIST) GenTreeArgList(impPopStack().val, tmp);
```
Comment (tannergooding) [inline src/coreclr/src/jit/hwintrinsicarm64.cpp:399]: Should there be a `gtNewArgList` overload that does this instead? (I'm not clear on when a helper method vs a direct `new (this, genTreeOps)` should be used)

From a quick grep, are there any other similar codepaths that might be missing flag propagation?

**Comment by tannergooding** [comment]
CC. @CarolEidt, @echesakovMSFT

File: src/coreclr/src/jit/hwintrinsicarm64.cpp (hunk 1 above)
Comment (EgorBo) [reply src/coreclr/src/jit/hwintrinsicarm64.cpp:399]: @tannergooding there are `gtNewArgList` overloads but those wrap arguments to list, e.g. https://github.com/dotnet/runtime/blob/master/src/coreclr/src/jit/gentree.cpp#L6475 (note that `gtNewArgList` for `argX`)

Comment (EgorBo) [reply src/coreclr/src/jit/hwintrinsicarm64.cpp:399]: Ah, actually there is `Compiler::gtNewListNode(GenTree* op1, GenTreeArgList* op2)` 🤔

Comment (EgorBo) [reply src/coreclr/src/jit/hwintrinsicarm64.cpp]: >  are there any other similar codepaths that might be missing flag propagation?

I couldn't find any

**Comment by AndyAyersMS** [comment]
There is a flag consistency checker `fgDebugCheckFlags` usually invoked via `fgDebugCheckLinks` that would have caught this, but we don't run it after the early phases. Perhaps we should try getting it enabled more broadly?

**Comment by EgorBo** [comment]
> There is a flag consistency checker `fgDebugCheckFlags` usually invoked via `fgDebugCheckLinks` that would have caught this, but we don't run it after the early phases. Perhaps we should try getting it enabled more broadly?

This specific issue got caught by `fgDebugCheckInlineCandidates` 🙂

**Comment by AndyAyersMS** [comment]
Right, but other flags might also not be accurate. We may do enough tree splitting with HW intrinsics that this rarely trips us up, but if we have reliable ways of creating interesting leaves in trees with HW intrinsic nodes, we may see other bad effects, eg inadvertently reordering intrinsic and other computation.

**Comment by EgorBo** [comment]
@AndyAyersMS just checked, if I disable `fgDebugCheckInlineCandidates` - it's caught by `fgDebugCheckFlags` then as you expected as far as I understand:
```
Extra flags on tree [000016]: -C---------
               [000016] --C--+------              *  RET_EXPR  float (inl return from call [000014])

Assert failure(PID 23388 [0x00005b5c], Thread: 41868 [0xa38c]): Assertion failed '!"Extra flags on tree"' in 'GitHub_43569.Program:Vector128_Create_float(float):float' during 'Morph - Global' (IL size 55)
```
(after Global Morph phase)

**Comment by erozenfeld** [comment]
Currently `fgDebugCheckFlags` starts checking only after morph. There are various issues before morph. I believe @sandreenko tried to enable this checking after importer at some point.

**Comment by EgorBo** [comment]
Ironically, Mono doesn't pass my test I added in this PR 😐

**Review by AndyAyersMS — APPROVED** [review]
LGTM. Thanks!

**Comment by AndyAyersMS** [comment]
We should port this to 5.0.

**Pull request merged**